gems/sorbet-runtime/lib/types/private/methods/call_validation.rb

```diff
@@ -12,7 +12,9 @@ module T::Private::Methods::CallValidation
   # @param method_sig [T::Private::Methods::Signature]
   # @return [UnboundMethod] the new wrapper method (or the original one if we didn't wrap it)
   def self.wrap_method_if_needed(mod, method_sig, original_method)
-    original_visibility = visibility_method_name(mod, method_sig.method_name)
+    # Use `original_method.name` not `method_sig.method_name` to get visibility
+    # so that we handle aliases correctly (we want the visibility before aliasing).
+    original_visibility = visibility_method_name(mod, original_method.name)
```

Comment:
Just checking my understanding: was this previously getting the visibility for the original method, instead fo the visibility of the alias? So for this example:
```ruby
class A
  def foo
    puts "hi"
  end

  alias :bar :foo
  private :bar

  def test
    bar
  end
end

A.new.test
```
Would we have used `public` as the visibility for the `:bar` alias, instead of `private`?

Reply:
This was previously getting the visibility of the original method, and continues to be getting the visibility of the original method. It probably should be getting the visibility of the alias, but we'd need to thread the aliasing module through, since that won't necessarily be the same as the `mod` here which is the original module.

But now that I'm looking at this I'm getting confused, I don't know how this is working at all with `mod` as the original module...

Reply:
Do we already have a test case for aliases where the modules differ?

Reply:
We do, but we weren't asserting that we got the fast path there. Now we are (and I fixed the bug, see other comment)